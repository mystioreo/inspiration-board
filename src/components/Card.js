import React, { Component } from 'react';
import PropTypes from 'prop-types';
import emoji from 'emoji-dictionary';

import './Card.css';

class Card extends Component {
  render() {
    return (
      <div key={this.props.id} className="card">
        <section className="card__content">
          <section className="card__content-text">
            {this.props.text}
          </section>
          <section className="card__content-emoji">
            {this.props.emoji && <p>{emoji.getUnicode(this.props.emoji) ? emoji.getUnicode(this.props.emoji) : this.props.emoji}</p>}
          </section>
        </section>
      </div>
    )
  }
}

Card.propTypes = {
  id: PropTypes.number.isRequired,
  text: PropTypes.string.isRequired,
  emoji: PropTypes.string,
};

export default Card;
