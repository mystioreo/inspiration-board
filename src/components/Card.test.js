import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';


describe('card', () => {
  it('will match the Card Snapshot', () => {
    const wrapper = shallow( <Card
                                id={1}
                                text="You Complete Me"
                                emoji="smile"
                                /> );

    expect(wrapper).toMatchSnapshot();
  })
});
