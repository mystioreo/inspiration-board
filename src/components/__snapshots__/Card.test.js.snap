// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card will match the Card Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    emoji="smile"
    id={1}
    text="You Complete Me"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "1",
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="card__content"
      >
        <section
          className="card__content-text"
        >
          You Complete Me
        </section>
        <section
          className="card__content-emoji"
        >
          <p>
            ðŸ˜„
          </p>
        </section>
      </section>,
      "className": "card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="card__content-text"
          >
            You Complete Me
          </section>,
          <section
            className="card__content-emoji"
          >
            <p>
              ðŸ˜„
            </p>
          </section>,
        ],
        "className": "card__content",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "You Complete Me",
            "className": "card__content-text",
          },
          "ref": null,
          "rendered": "You Complete Me",
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              ðŸ˜„
            </p>,
            "className": "card__content-emoji",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "ðŸ˜„",
            },
            "ref": null,
            "rendered": "ðŸ˜„",
            "type": "p",
          },
          "type": "section",
        },
      ],
      "type": "section",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "1",
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="card__content"
        >
          <section
            className="card__content-text"
          >
            You Complete Me
          </section>
          <section
            className="card__content-emoji"
          >
            <p>
              ðŸ˜„
            </p>
          </section>
        </section>,
        "className": "card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <section
              className="card__content-text"
            >
              You Complete Me
            </section>,
            <section
              className="card__content-emoji"
            >
              <p>
                ðŸ˜„
              </p>
            </section>,
          ],
          "className": "card__content",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "You Complete Me",
              "className": "card__content-text",
            },
            "ref": null,
            "rendered": "You Complete Me",
            "type": "section",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                ðŸ˜„
              </p>,
              "className": "card__content-emoji",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ðŸ˜„",
              },
              "ref": null,
              "rendered": "ðŸ˜„",
              "type": "p",
            },
            "type": "section",
          },
        ],
        "type": "section",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
